lms/djangoapps/courseware/views.py

```diff
@@ -894,6 +895,18 @@ def course_about(request, course_id):
             reg_then_add_to_cart_link = "{reg_url}?course_id={course_id}&enrollment_action=add_to_cart".format(
                 reg_url=reverse('register_user'), course_id=urllib.quote(str(course_id)))
 
+        ecommerce_checkout_link = ''
+        mode = ''
+        ecomm_service = EcommerceService()
+        if ecomm_service.is_enabled() and (
+            'professional' in modes or 'no-id-professional' in modes
+        ):
+            mode = modes.get('professional', '') or modes.get('no-id-professional', '')
+            ecommerce_checkout_link = "{}?sku={}" .format(ecomm_service.payment_page_url(), mode.sku)
```

Comment:
Should the EcommerceService own the construction of the checkout link, versus having this logic in the view?  Then it could be used in more places.